src/core/Akka.Streams/Implementation/Throttle.cs

```diff
@@ -29,35 +31,62 @@ private sealed class Logic : TimerGraphStageLogic, IInHandler, IOutHandler
             private readonly Throttle<T> _stage;
             private readonly TickTimeTokenBucket _tokenBucket;
             private readonly bool _enforcing;
+            private readonly Decider _decider;
 
             private bool _willStop;
             private Option<T> _currentElement;
 
-            public Logic(Throttle<T> stage) : base(stage.Shape)
+            public Logic(Throttle<T> stage, Attributes inheritedAttributes) : base(stage.Shape)
             {
                 _stage = stage;
                 _tokenBucket = new TickTimeTokenBucket(stage._maximumBurst, stage._ticksBetweenTokens);
                 _enforcing = stage._mode == ThrottleMode.Enforcing;
 
+                var attr = inheritedAttributes.GetAttribute<ActorAttributes.SupervisionStrategy>(null);
+                _decider = attr != null ? attr.Decider : Deciders.StoppingDecider;
+                
                 SetHandler(_stage.Inlet, this);
                 SetHandler(_stage.Outlet, this);
             }
 
             public void OnPush()
             {
                 var element = Grab(_stage.Inlet);
-                var cost = _stage._costCalculation(element);
-                var delayTicks = _tokenBucket.Offer(cost);
-
-                if (delayTicks == 0)
-                    Push(_stage.Outlet, element);
-                else
+                try
                 {
-                    if (_enforcing)
-                        throw new OverflowException("Maximum throttle throughput exceeded.");
-
-                    _currentElement = element;
-                    ScheduleOnce(TimerName, TimeSpan.FromTicks(delayTicks));
+                    var cost = _stage._costCalculation(element);
+                    var delayTicks = _tokenBucket.Offer(cost);
+
+                    if (delayTicks == 0)
+                        Push(_stage.Outlet, element);
+                    else
+                    {
+                        if (_enforcing)
+                            throw new OverflowException("Maximum throttle throughput exceeded.");
```

Comment:
Problem: if we're enforcing, do we need to fail regardless of what the supervision strategy says?

Reply:
I'm not sure, what do you think is best?

Reply:
I think it should fail regardless, I'll change the code

Reply:
I agree